nab/detectors/numenta/numentaTM_detector.py

```diff
@@ -0,0 +1,109 @@
+# ----------------------------------------------------------------------
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import math
+import simplejson as json
+
+from nupic.algorithms import anomaly_likelihood
+from nupic.frameworks.opf.common_models.cluster_params import (
+  getScalarMetricWithTimeOfDayAnomalyParams)
+from nupic.frameworks.opf.modelfactory import ModelFactory
+
+from nab.detectors.base import AnomalyDetector
+
+
+
+class NumentaTMDetector(AnomalyDetector):
```

Comment:
Yes, the only difference is in the initialize function. I'll make it a subclass of NumentaDetector and add comments.